python/lsst/daf/butler/direct_butler/_direct_butler.py

```diff
@@ -2076,86 +2047,126 @@ def transfer_from(
             dimension_records = self._extract_all_dimension_records_from_data_ids(
                 source_butler, dataIds, elements
             )
+        return grouped_refs, dimension_records, artifact_existence
 
+    def _import_dimension_records(
+        self, dimension_records: dict[DimensionElement, dict[DataCoordinate, DimensionRecord]], dry_run: bool
+    ) -> None:
+        """Import dimension records collected during import pre-process."""
+        if dimension_records and not dry_run:
+            _LOG.verbose("Ensuring that dimension records exist for transferred datasets.")
+            # Order matters.
+            for element in self.dimensions.sorted(dimension_records.keys()):
+                records = [r for r in dimension_records[element].values()]
```

Comment:
Could be just `list(dimension_records[element].values())`?

Reply:
Yes. Thanks. This was old code that was moved.